test/e2e/context/context_k8s_lifecycle_test.go

```diff
@@ -0,0 +1,98 @@
+		})
+		// Test case: (negative test) Create context for k8s target with incorrect kubeconfig file path and its context as input
+		It("create context with incorrect kubeconfig and context", func() {
+			ctxName := ContextNameConfigPrefix + framework.RandomString(4)
+			err := tf.ContextCmd.CreateConextWithKubeconfig(ctxName, framework.RandomString(4), clusterInfo.ClusterContext)
+			Expect(err).ToNot(BeNil())
+			Expect(strings.Contains(err.Error(), framework.FailedToCreateContext)).To(BeTrue())
+		})
+		// Test case: (negative test) Create context for k8s target with kubeconfig file path and incorrect context as input
+		It("create context with kubeconfig and incorrect context", func() {
+			ctxName := ContextNameConfigPrefix + framework.RandomString(4)
+			err := tf.ContextCmd.CreateConextWithKubeconfig(ctxName, clusterInfo.KubeConfigPath, framework.RandomString(4))
+			Expect(err).ToNot(BeNil())
+			Expect(strings.Contains(err.Error(), framework.FailedToCreateContext)).To(BeTrue())
+		})
+		// Test case: Create context for k8s target with "default" kubeconfig and its context only as input value
+		It("create context with kubeconfig and context", func() {
+			ctxName := "context-defaultConfig-" + framework.RandomString(4)
+			err := tf.ContextCmd.CreateContextWithDefaultKubeconfig(ctxName, clusterInfo.ClusterContext)
+			Expect(err).To(BeNil(), "context should create without any error")
+			contextNames = append(contextNames, ctxName)
+			active, err := tf.ContextCmd.GetActiveContext(framework.TargetTypeK8s)
+			Expect(err).To(BeNil(), "there should be a active context")
+			Expect(active).To(Equal(ctxName), "the active context should be recently added context")
+		})
+		// Test case: test 'tanzu context use' command with the specific context name (not the recently created one)
+		It("use context command", func() {
+			err := tf.ContextCmd.UseContext(contextNames[0])
+			Expect(err).To(BeNil(), "use context should set context without any error")
+			active, err := tf.ContextCmd.GetActiveContext(framework.TargetTypeK8s)
+			Expect(err).To(BeNil(), "there should be a active context")
+			Expect(active).To(Equal(contextNames[0]), "the active context should be recently set context")
+		})
+		// Test case: (negative test) test 'tanzu context use' command with the specific context name (incorrect, which is not exists)
+		It("use context command with incorrect context as input", func() {
+			err := tf.ContextCmd.UseContext(framework.RandomString(4))
+			Expect(err).ToNot(BeNil())
+		})
+		// Test case: test 'tanzu context list' command, should list all contexts created
+		It("list context should have all added contexts", func() {
+			list := GetAvailableContexts(tf, contextNames)
+			Expect(len(list)).To(Equal(len(contextNames)), "list context should have all contexts added in previous tests")
+		})
+		// Test case: test 'tanzu context delete' command, make sure to delete all context's created in previous test cases
```

Comment:
Let's add one negative test case for `tanzu context delete` where the context specified doesn't exist.

Reply:
added